compiler/rustc_hir_typeck/src/upvar.rs

```diff
@@ -1195,15 +1190,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     if let Some(reasons) = auto_trait_diagnostic.get(&captures_info) {
                         reasons.clone()
                     } else {
-                        FxHashSet::default()
+                        UnordSet::default()
                     };
 
                 // Check if migration is needed because of drop reorder as a result of that capture
                 let capture_drop_reorder_reason = drop_reorder_diagnostic.contains(&captures_info);
 
                 // Combine all the reasons of why the root variable should be captured as a result of
                 // auto trait implementation issues
-                auto_trait_migration_reasons.extend(capture_trait_reasons.iter().copied());
+                auto_trait_migration_reasons.extend_unord(capture_trait_reasons.items().copied());
```

Comment:
I'll take a look if I can rename it without that leading to clashes with `Extend::extend`.

Reply:
Calling the method `extend` instead of `extend_unord` indeed seems to require explicit disambiguation in some cases. So instead, I tried to implement it more cleanly by adding an `ExtendUnord` trait in https://github.com/rust-lang/rust/pull/113328/commits/7587806b9dff8efa1e72a092327409795431f92c.